src/main/java/org/example/model/Product.java

```diff
@@ -0,0 +1,46 @@
+package org.example.model;
+
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class Product {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private String name;
+  private String description;
+  private double price;
+  private String currency;
+
+  public Product(Long id, String name, String description, double price, String currency) {
```

Comment:
Use the [Currency class here. ](https://www.baeldung.com/java-money-and-currency)

Instead of `double`, use [BigDecimal.](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html)